core/dbt/deps/base.py

```diff
@@ -97,6 +99,48 @@ def get_installation_path(self, project, renderer):
     def get_subdirectory(self):
         return None
 
+    def _install(self, project, renderer):
+        """
+        verbatim from dbt.deps.registry.RegistryPinnedPackage
+            install() from RegistryPinnedPackage class
+            Q for dbt labs, is this an appropriate pattern to refactor against?
```

Comment:
Questions for dbt labs, is this an appropriate pattern to refactor against?
Fns used upstream in RegistryPinnedPackage and TarballPinnedPackage via inheritance (originally from from dbt.deps.registry.RegistryPinnedPackage)

Reply:
Sure is!  Everywhere else that inherits from `PinnedPackage` has a custom `install` method.

Reply:
ty for feedback, will close this comment and remove from code